fuzzer/src/fuzz_json.rs

```diff
@@ -1,165 +1,23 @@
-    #[error(transparent)]
-    EncodeTrace(#[from] EncodeTraceError),
-    #[error(transparent)]
-    VirtualMachine(#[from] VirtualMachineError),
-    #[error(transparent)]
-    Trace(#[from] TraceError),
-}
-
-struct FileWriter {
-    buf_writer: io::BufWriter<std::fs::File>,
-    bytes_written: usize,
-}
-
-impl Writer for FileWriter {
-    fn write(&mut self, bytes: &[u8]) -> Result<(), bincode::error::EncodeError> {
-        self.buf_writer
-            .write_all(bytes)
-            .map_err(|e| bincode::error::EncodeError::Io {
-                inner: e,
-                index: self.bytes_written,
-            })?;
-
-        self.bytes_written += bytes.len();
-
-        Ok(())
-    }
-}
-
-impl FileWriter {
-    fn new(buf_writer: io::BufWriter<std::fs::File>) -> Self {
-        Self {
-            buf_writer,
-            bytes_written: 0,
-        }
-    }
-
-    fn flush(&mut self) -> io::Result<()> {
-        self.buf_writer.flush()
-    }
-}
-
-fn run(args: Args) -> Result<(), Error> {
-    let trace_enabled = args.trace_file.is_some();
-    let mut hint_executor = BuiltinHintProcessor::new_empty();
-    let cairo_run_config = cairo_run::CairoRunConfig {
-        entrypoint: &args.entrypoint,
-        trace_enabled,
-        relocate_mem: args.memory_file.is_some(),
-        layout: &args.layout.to_string(),
-        proof_mode: args.proof_mode,
-        secure_run: args.secure_run,
-    };
-
-    let (cairo_runner, mut vm) =
-        match cairo_run::cairo_run(&args.program_content, &cairo_run_config, &mut hint_executor) {
-            Ok(runner) => runner,
-            Err(error) => {
-                eprintln!("{error}");
-                return Err(Error::Runner(error));
-            }
-        };
-
-    if args.print_output {
-        let mut output_buffer = "Program Output:\n".to_string();
-        vm.write_output(&mut output_buffer)?;
-        print!("{output_buffer}");
-    }
-
-    if let Some(trace_path) = args.trace_file {
-        let relocated_trace = vm.get_relocated_trace()?;
-
-        let trace_file = std::fs::File::create(trace_path)?;
-        let mut trace_writer =
-            FileWriter::new(io::BufWriter::with_capacity(3 * 1024 * 1024, trace_file));
-
-        cairo_run::write_encoded_trace(relocated_trace, &mut trace_writer)?;
-        trace_writer.flush()?;
-    }
-
-    if let Some(memory_path) = args.memory_file {
-        let memory_file = std::fs::File::create(memory_path)?;
-        let mut memory_writer =
-            FileWriter::new(io::BufWriter::with_capacity(5 * 1024 * 1024, memory_file));
-
-        cairo_run::write_encoded_memory(&cairo_runner.relocated_memory, &mut memory_writer)?;
-        memory_writer.flush()?;
-    }
-
-    Ok(())
-}
 
 fn main() {
     loop {
-        fuzz!(|args: Args| {
-            let _ = run(args);
+        fuzz!(|data: (CairoRunConfig, &[u8])| {
+            let (cairo_run_config, program_json) = data;
+            let _ = cairo_run(
```

Comment:
Using `cairo_run` instead of `cairo_run_fuzzed_program` due to the unlikeliness of the fuzzer creating an infinite loop and wanting to get as close to normal VM execution as possible